### PR TITLE
feat(plugin): Ify improvements

### PR DESCRIPTION
**Pull request by Nuckyz**
Improve Spotify premium bypass, turn disable auto pause into an option and add option to keep the activity playing when idling

### CODE REVIEW

**Comment by Ruthenic** [comment]
Why in the hell would you want to re-enable autopausing

**Comment by Nuckyz** [comment]
> Why in the hell would you want to re-enable autopausing

It doesn't hurt to have it as an option

File: src/plugins/ify.ts [hunk 1]
```diff
@@ -1,18 +1,47 @@
...
+    ],
     patches: [{
-        find: '.displayName="SpotifyStore"',
+        find: 'dispatch({type:"SPOTIFY_PROFILE_UPDATE"',
         replacement: [{
-            match: /\.isPremium=.;/,
-            replace: ".isPremium=true;"
-        }, {
+            match: /(function\((.{1,2})\){)(.{1,6}dispatch\({type:"SPOTIFY_PROFILE_UPDATE")/,
+            replace: '$1$2.body.product=\"premium\";$3'
```
Comment (Vendicated) [inline src/plugins/ify.ts:16]: what's the point of changing this patch?

Comment (Nuckyz) [reply src/plugins/ify.ts:16]: I haven't found any difference but that patches the very source of where isPremium comes from, seems better to avoid issues

Comment (Vendicated) [reply src/plugins/ify.ts:16]: I disagree, it's much more complex than the old one so I would prefer the old one

File: src/plugins/ify.ts [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
+        predicate: () => Settings.plugins["Ify"].noSpotifyAutoPause === true,
+        replacement: {
             match: /function (.{1,2})\(\).{0,200}SPOTIFY_AUTO_PAUSED\);.{0,}}}}/,
             replace: "function $1(){}"
-        }]
-    }]
+        }
+    }, {
+        find: '.displayName="SpotifyStore"',
+        predicate: () => Settings.plugins["Ify"].keepSpotifyActivityOnIdle === true,
```
Comment (Vendicated) [inline src/plugins/ify.ts:27]: please prefer direct property access (Settings.plugins.Ify) and remove the comparison to true, it is obsolete

Comment (Nuckyz) [reply src/plugins/ify.ts:27]: I took inspiration from the experiments plugin, that's why it's like, gonna change it

Comment (Vendicated) [reply src/plugins/ify.ts:27]: Then that should be changed as well (not in this PR). Maybe there's an eslint rule which can enforce this in the future? 🤔

**Comment by Vendicated** [comment]
could you please change the patches from using $1$2 to using a function?
`replace: (fullMatch, firstGroup, secondGroup) => ""`
this way its more readable since thanks to named groups you don't have to decipher what each group stands for

File: src/plugins/ify.ts (hunk 1 above)
Comment (Nuckyz) [reply src/plugins/ify.ts]: Can this be considered resolved?

**Review by Vendicated — APPROVED** [review]

**Pull request merged**